### ISSUE TITLE
Debug why we can't handle this csv file

### ISSUE THREAD

**Issue by scsmithr** [GlareDB/glaredb#1707]
## Description

Link here: https://clo-pfw-prod.s3.us-west-2.amazonaws.com/data/202306/PFW_all_2021_2023_June2023_Public.zip

```
> select * from './birds.csv' limit 10;
Error: Arrow error: Parser error: Error while parsing value NA for column 13 at line 132
```

**Comment by vrongmeal** [comment]
The reason is that `NA` doesn't parse as `NULL` for int columns.

**Comment by vrongmeal** [comment]
[EDIT] This is strange...

**Comment by vrongmeal** [comment]
Oh okay! Yeah, it's basically because `NA` isn't inferred as `NULL`. The schema is inferred as int column and it finds `NA` so it errors.

**Comment by vrongmeal** [comment]
The initial few rows of data are valid integers for the mentioned column. What can help here is increasing the number of rows that are used to infer schema, but that would fail as well.

What we really need is an option where we can tell the file format to infer and execute and consider `NA` as nulls. Maybe we should think about implementing our own `FileFormat` on top of `CsvFormat`.

**Comment by vrongmeal** [comment]
We already do infer from a lot of rows:

```
        let mut records_to_read = self.schema_infer_max_rec.unwrap_or(usize::MAX);
```

This file just happens to not have any `NA` in these many rows.

**Comment by vrongmeal** [comment]
Temporary fix in #1726

**Comment by vrongmeal** [comment]
Opened https://github.com/apache/arrow-rs/pull/4795
